libs/application/ui-fields/src/lib/TableRepeaterFormField/TableRepeaterFormField.tsx

```diff
@@ -372,3 +284,190 @@ export const TableRepeaterFormField: FC<Props> = ({
+     * This hack is needed to get the correct type
+     */
+    const errorList = error as unknown as Record<string, string>[] | undefined
+    const errors = errorList?.[activeIndex]
+    return errors?.[id]
+  }
+
+  const getDefaultValue = (
+    item: TableRepeaterItem,
+    application: Application,
+    activeField?: Record<string, string>,
+  ) => {
+    const { defaultValue } = item
+
+    if (defaultValue === undefined) {
+      return undefined
+    }
+
+    return defaultValue(application, activeField)
+  }
+
+  let translatedOptions: any = []
+  if (typeof options === 'function') {
+    translatedOptions = options(application, activeValues)
+  } else {
+    translatedOptions = options?.map((option) => ({
+      ...option,
+      label: formatText(option.label, application, formatMessage),
+      ...(option.tooltip && {
+        tooltip: formatText(option.tooltip, application, formatMessage),
+      }),
+    }))
+  }
+
+  let Readonly: boolean | undefined
+  if (typeof readonly === 'function') {
+    Readonly = readonly(application, activeValues)
+  } else {
+    Readonly = readonly
+  }
+
+  let DefaultValue: any
+  if (component === 'input') {
+    DefaultValue = getDefaultValue(item, application, activeValues)
+  }
+  if (component === 'select') {
+    DefaultValue =
+      getValueViaPath(application.answers, id) ??
+      getDefaultValue(item, application, activeValues)
+  }
+  if (component === 'radio') {
+    DefaultValue =
+      (getValueViaPath(application.answers, id) as string[]) ??
+      getDefaultValue(item, application, activeValues)
+  }
+  if (component === 'checkbox') {
+    DefaultValue =
+      (getValueViaPath(application.answers, id) as string[]) ??
+      getDefaultValue(item, application, activeValues)
+  }
+  if (component === 'date') {
+    DefaultValue =
+      (getValueViaPath(application.answers, id) as string) ??
+      getDefaultValue(item, application, activeValues)
+  }
+
+  if (condition && !condition(application, activeValues)) {
+    return null
+  }
+
+  return (
+    <GridColumn span={['1/1', '1/1', '1/1', span]}>
+      {component === 'radio' && label && (
+        <Text variant="h4" as="h4" id={id + 'title'} marginBottom={3}>
+          {formatText(label, application, formatMessage)}
+        </Text>
+      )}
+      <Component
+        id={id}
+        name={id}
+        label={formatText(label, application, formatMessage)}
+        options={translatedOptions}
+        placeholder={formatText(placeholder, application, formatMessage)}
+        split={width === 'half' ? '1/2' : '1/1'}
+        error={getFieldError(itemId)}
+        control={control}
+        readOnly={Readonly}
+        backgroundColor={backgroundColor}
+        onChange={() => {
+          if (error) {
+            clearErrors(id)
+          }
+        }}
+        application={application}
+        defaultValue={DefaultValue}
+        {...props}
+      />
+    </GridColumn>
+  )
+}
```

Comment:
I like these changes to the `TableRepeater`, but we should think about moving the `Item` component to a separate file, since this one is almost 500 lines.

Also we need to make sure that every other application that uses this field still works as expected, since this is a pretty big change

Reply:
we are the code owners of most of the applications that uses the table repeater and it works as expected :) there are two applications that we haven't tested that we think you (Norda) are the code owners of. Can you see if it works there?

- [ ] GrindavikHousingBuyoutForm
- [ ] HomeSupportForm
- [x] relativesSubSection
- [x] siblingsSubSection
- [x] EditOrAddEmployersAndPeriods
- [x] ParentalLeaveForm
- [x] IncomePlanForm
- [-] Signature collection (table repeater commented out)